pkg/util/limitrange/limitrange_test.go

```diff
@@ -161,90 +160,184 @@ func TestSummarize(t *testing.T) {
+		"pod without init containers": {
 			podSpec: &corev1.PodSpec{
-				Containers: containers[:2],
+				Containers: []corev1.Container{
+					*testingutil.MakeContainer().
+						WithResourceReq(corev1.ResourceCPU, "1").
+						WithResourceReq(corev1.ResourceMemory, "2Gi").
+						Obj(),
+					*testingutil.MakeContainer().
+						WithResourceReq(corev1.ResourceCPU, "1500m").
+						WithResourceReq("example.com/gpu", "2").
+						Obj(),
+				},
 			},
 			want: corev1.ResourceList{
 				corev1.ResourceCPU:    resource.MustParse("2500m"),
 				corev1.ResourceMemory: resource.MustParse("2Gi"),
 				"example.com/gpu":     resource.MustParse("2"),
 			},
 		},
-		"one init wants more": {
+		"pod only with regular init containers": {
 			podSpec: &corev1.PodSpec{
-				InitContainers: containers[2:4],
-				Containers:     containers[:2],
+				InitContainers: []corev1.Container{
+					*testingutil.MakeContainer().
+						WithResourceReq(corev1.ResourceCPU, "4").
+						WithResourceReq(corev1.ResourceMemory, "2Gi").
+						Obj(),
+					*testingutil.MakeContainer().
+						WithResourceReq(corev1.ResourceCPU, "1500m").
+						WithResourceReq("example.com/gpu", "2").
+						Obj(),
+				},
+				Containers: []corev1.Container{
+					*testingutil.MakeContainer().
+						WithResourceReq(corev1.ResourceCPU, "1").
+						WithResourceReq(corev1.ResourceMemory, "2Gi").
+						Obj(),
+					*testingutil.MakeContainer().
+						WithResourceReq(corev1.ResourceCPU, "1500m").
+						WithResourceReq("example.com/gpu", "2").
+						Obj(),
+				},
 			},
 			want: corev1.ResourceList{
 				corev1.ResourceCPU:    resource.MustParse("4000m"),
 				corev1.ResourceMemory: resource.MustParse("2Gi"),
 				"example.com/gpu":     resource.MustParse("2"),
 			},
 		},
-		"include sidecar container": {
+		"pod only with sidecar containers": {
 			podSpec: &corev1.PodSpec{
-				InitContainers: containers[2:],
-				Containers:     containers[:2],
+				InitContainers: []corev1.Container{
+					*testingutil.MakeContainer().
+						WithResourceReq(corev1.ResourceCPU, "4").
+						WithResourceReq(corev1.ResourceMemory, "2Gi").
+						AsSidecar().
+						Obj(),
+					*testingutil.MakeContainer().
+						WithResourceReq(corev1.ResourceCPU, "1500m").
+						WithResourceReq("example.com/gpu", "2").
+						AsSidecar().
+						Obj(),
+				},
+				Containers: []corev1.Container{
+					*testingutil.MakeContainer().
+						WithResourceReq(corev1.ResourceCPU, "1").
+						WithResourceReq(corev1.ResourceMemory, "2Gi").
+						Obj(),
+					*testingutil.MakeContainer().
+						WithResourceReq(corev1.ResourceCPU, "1500m").
+						WithResourceReq("example.com/gpu", "2").
+						Obj(),
+				},
 			},
 			want: corev1.ResourceList{
-				corev1.ResourceCPU:    resource.MustParse("5000m"),
+				corev1.ResourceCPU:    resource.MustParse("8000m"),
 				corev1.ResourceMemory: resource.MustParse("4Gi"),
 				"example.com/gpu":     resource.MustParse("4"),
 			},
 		},
+		"pod only with init containers": {
```

Comment:
there are sidecars in this test?

Reply:
Yes, the last two items in `InitContainers` field are sidecars, and the first two are regular init containers.  I updated the description for more clarity.